### PR TITLE
changing Type in ast.proto to make it consistent with the datasets

### PR DESCRIPTION
**Pull request by gupadhyaya**
This pull request changes the Type "name" field from int32 to string and comments out "qualified_name" field in the Type that did not exist while generating the datasets.

### CODE REVIEW

**Comment by hridesh** [comment]
CI is failing. Also, please clarify whether this was tested on a compatible dataset.

**Comment by gupadhyaya** [comment]
does this mean, we need to change all the relevant test cases?

**Comment by psybers** [comment]
Should only be datagen tests failing. It would be very painful to fix those... Maybe we should temporarily remove them to the broken test folder?

File: src/java/boa/datagen/util/Java7Visitor.java [hunk 1]
```diff
@@ -179,19 +179,19 @@ public boolean visit(TypeDeclaration node) {
 			}
 			if (bounds.length() > 0)
 				name = name + " extends " + bounds;
-			tb.setName(getIndex(name));
```
Comment (psybers) [inline src/java/boa/datagen/util/Java7Visitor.java]: You can remove the getIndex() method at the end of the class too.

**Comment by psybers** [comment]
Actually we need the proto to be this exact one: https://github.com/boalang/compiler/blob/compiler-2015-08/src/proto/ast.proto can you please update to this (and regenerate the Ast.java)?

**Comment by gupadhyaya** [comment]
I had to add lambda stuffs to make Java8Visitor compile, if that is okay?

**Comment by psybers** [comment]
I think it should be ok.  Protobufs are fine with changes as long as the change is adding new fields and none of the field IDs were modified.  I am going to merge this now.  If it fails for some reason, I'll go ahead and fix it.

**Pull request merged**